### PR TITLE
feat(efs): add EFS file system policy and update module source

### PR DESCRIPTION
**Pull request by sachincool**
Introduce a new EFS file system policy to enhance security by
defining specific access permissions. Update the EFS module source
to "cloudposse/efs/aws" for improved compatibility and features.
Add a new variable for region configuration to allow flexible
deployment across different AWS regions.

### CODE REVIEW

File: efs.tf [hunk 1]
```diff
@@ -62,44 +68,42 @@ data "aws_iam_policy_document" "efs" {
...
-  performance_mode     = var.performance_mode
-  enable_backup_policy = var.enable_backup_policy
-  security_group_rules = {
-    vpc = {
-      # relying on the defaults provdied for EFS/NFS (2049/TCP + ingress)
-      description = "NFS ingress from VPC private subnets"
-      cidr_blocks = var.private_subnets_cidrs
-    }
-  }
+  # efs_file_system_policy             = data.aws_iam_policy_document.efs_file_system_policy.json
```
Comment (shubhamrai1993) [inline efs.tf]: shouldn't be commented right

File: efs.tf [hunk 2]
```diff
@@ -4,6 +4,14 @@ resource "aws_iam_policy" "efs" {
   policy      = data.aws_iam_policy_document.efs.json
   tags        = local.tags
 }
+
+resource "aws_efs_file_system_policy" "policy" {
```
Comment (shubhamrai1993) [inline efs.tf]: Why this resource?

File: efs.tf [hunk 3]
```diff
@@ -62,44 +68,42 @@ data "aws_iam_policy_document" "efs" {
...
-        test     = "Bool"
-        values   = ["true"]
-        variable = "elasticfilesystem:AccessedViaMountTarget"
-      }]
-    }
-  ]
-  throughput_mode      = var.throughput_mode
-  performance_mode     = var.performance_mode
-  enable_backup_policy = var.enable_backup_policy
-  security_group_rules = {
```
Comment (shubhamrai1993) [inline efs.tf:96]: These security group rules don't need to be specified in the other module?

Comment (sachincool) [reply efs.tf:96]: it is specified in line number 74

File: efs.tf (hunk 2 above)
Comment (sachincool) [reply efs.tf]: cloudposse was also creating this resource,but it relied on count( without data block)
but the argument of this requires a `policy` which needs to be generated via data block.

example: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/efs_file_system_policy

we had to move this out of that module, to get smooth one-shot plan 
https://github.com/cloudposse/terraform-aws-efs/blob/main/main.tf#L153-L159

File: efs.tf (hunk 1 above)
Comment (sachincool) [reply efs.tf]: have resolved this

File: UPGRADE.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+### Breaking Changes
+
+1. The EFS filesystem resource name has been changed from [aws_efs_module](https://registry.terraform.io/modules/terraform-aws-modules/efs/aws/latest) to [cloudposse_efs](https://registry.terraform.io/modules/cloudposse/efs/aws/latest), due to count dependency on data block inputs. This helps us generate a one-shot plan for the whole Truefoundry installation.
+
+### Required Actions
+
+Before running `terraform plan`, you need to migrate the state of the EFS filesystem resource to prevent destruction and recreation. Execute the following command:
+
+```bash
+terraform state mv 'module.efs.module.efs.aws_efs_file_system.default[0]' 'module.efs.module.efs.aws_efs_file_system.this[0]'
```
Comment (dunefro) [inline UPGRADE.md]: there should be more terraform state mv commands, isn't it ?

**Review by shubhamrai1993 — APPROVED** [review]

**Pull request merged**